examples/job/work-queue-1/README.md

```diff
@@ -0,0 +1,339 @@
+
+You'll need to make a copy of the Job and edit the image to match the name you used, and call it
+`./job-wq-1.yaml`.
+
+## Running the Job
+
+So, now run the Job:
+
+```console
+$ kubectl create -f ./job-wq-1.yaml
+```
+
+Now wait a bit, then check on the job.
+
+```console
+$ ./kubectl describe jobs/job-wq-1 
+Name:		job-wq-1
+Namespace:	default
+Image(s):	gcr.io/causal-jigsaw-637/job-wq-1
+Selector:	app in (job-wq-1)
+Parallelism:	4
+Completions:	8
+Labels:		app=job-wq-1
+Pods Statuses:	0 Running / 8 Succeeded / 0 Failed
+No volumes.
+Events:
+  FirstSeen	LastSeen	Count	From	SubobjectPath	Reason			Message
+  ─────────	────────	─────	────	─────────────	──────			───────
+  27s		27s		1	{job }			SuccessfulCreate	Created pod: job-wq-1-hcobb
+  27s		27s		1	{job }			SuccessfulCreate	Created pod: job-wq-1-weytj
+  27s		27s		1	{job }			SuccessfulCreate	Created pod: job-wq-1-qaam5
+  27s		27s		1	{job }			SuccessfulCreate	Created pod: job-wq-1-b67sr
+  26s		26s		1	{job }			SuccessfulCreate	Created pod: job-wq-1-xe5hj
+  15s		15s		1	{job }			SuccessfulCreate	Created pod: job-wq-1-w2zqe
+  14s		14s		1	{job }			SuccessfulCreate	Created pod: job-wq-1-d6ppa
+  14s		14s		1	{job }			SuccessfulCreate	Created pod: job-wq-1-p17e0
+```
+
+All our pods succeeded.  Yay.
+
+
+## Alternatives
+
+If running a queue service or modifying your containers to use a work queue is inconvenient, you may
+want to consider one of the other [job patterns](../../../docs/user-guide/jobs.md#job-patterns).
+
+This approach creates a pod for every work item.  If your work items only take a few seconds,
+though, creating a Pod for every work item may add a lot of overhead.  Consider another
+[example](../work-queue-2/README.md), that executes multiple work items per Pod.
+
+In this example, we used use the `amqp-consume` utility to read the message
+from the queue and run our actual program.  This has the advantage that you
+do not need to modify your program to be aware of the queue.
+A [different example](../work-queue-2/README.md), shows how to
+communicate with the work queue using a client library.
+
+## Caveats
+
+If the number of completions is set to less than the number of items in the queue, then
+not all items will be processed.
+
+If the number of completions is set to more than the number of items in the queue,
+then the Job will not appear to be completed, even though all items in the queue
+have been processed.  It will start additional pods which will block waiting
+for a mesage.
+
+There is an unlikely race with this pattern.  If the container is killed in between the time
+that the message is acknowledged by the amqp-consume command and the time that the container
+exits with success, or if the node crashes before the kubelet is able to post the success of the pod
+back to the api-server, then the Job will not appear to be complete, even though all items
+in the queue have been processed.
```

Comment:
I'd add that possible solution to this might be ACK-ing after processing the item instead of before.